### PR TITLE
feat(deploy-task): added zupass gatekeeper to the deploy tasks

### PR DESCRIPTION
**Pull request by crisgarner**
# Description

 Added zupass gatekeeper to the deploy tasks and deploy config example

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Thanks, left some comments

**Comment by crisgarner** [comment]
Thanks! implemented the recommendations

File: contracts/tasks/deploy/maci/02-gatekeepers.ts [hunk 1]
```diff
@@ -110,5 +118,39 @@ deployment.deployTask("full:deploy-gatekeepers", "Deploy gatekeepers").then((tas
...
     }
+
+    if (!skipDeployZupassGatekeeper) {
+      const eventId = deployment.getDeployConfigField<string>(EContracts.ZupassGatekeeper, "eventId", true);
+      const validEventId = uuidToBigInt(eventId);
+      const validSigner1 = deployment.getDeployConfigField<string>(EContracts.ZupassGatekeeper, "signer1", true);
+      const validSigner2 = deployment.getDeployConfigField<string>(EContracts.ZupassGatekeeper, "signer2", true);
+      let verifier: string | undefined = deployment.getDeployConfigField<string>(
+        EContracts.ZupassGatekeeper,
+        "zupassVerifier",
```
Comment (ctrlc03) [inline contracts/tasks/deploy/maci/02-gatekeepers.ts:129]: should we then add this to the config template?

File: contracts/tasks/deploy/maci/02-gatekeepers.ts [hunk 2]
```diff
@@ -110,5 +118,39 @@ deployment.deployTask("full:deploy-gatekeepers", "Deploy gatekeepers").then((tas
...
+      );
+      if (!verifier) {
+        const verifierContract = await deployment.deployContract({
+          name: EContracts.Groth16Verifier,
```
Comment (ctrlc03) [inline contracts/tasks/deploy/maci/02-gatekeepers.ts:133]: let's change the contract name to be ZupassVerifier (or ZupassGroth16Verifier)

Comment (crisgarner) [reply contracts/tasks/deploy/maci/02-gatekeepers.ts:133]: Got it! thought it was only for the config variable and not confuse coordinators

**Review by 0xmad — CHANGES_REQUESTED** [review]
@crisgarner thanks, left some comments

**Comment by crisgarner** [comment]
Thanks! added the recommendations @0xmad

**Comment by ctrlc03** [comment]
> Thanks! added the recommendations @0xmad

seems like the project is not building

**Review by kittybest — APPROVED** [review]
Thanks, just need to check the import path / file nam.

**Review by ctrlc03 — APPROVED** [review]

**Review by 0xmad — APPROVED** [review]
@crisgarner thanks!

**Pull request merged**